console/src/telemetry/index.ts

```diff
@@ -0,0 +1,108 @@
+import endpoints from '../Endpoints';
+import globals from '../Globals';
+import { filterEventsBlockList, sanitiseUrl } from './filters';
+import { RUN_TIME_ERROR } from '../components/Main/Actions';
+
+type ReduxAction = {
+  type: string;
+  payload: {
+    pathname: string;
+  };
+  data: any;
+};
+
+interface TelemetryGlobals {
+  serverVersion: string;
+  consoleMode: string;
+  cliUUID: string;
+  hasuraUUID: string;
+}
+
+const createClient = () => {
+  if (globals.enableTelemetry) {
+    try {
+      const client = new WebSocket(endpoints.telemetryServer);
+      client.onerror = e => {
+        console.error(`WebSocket Error for Events${e}`);
+      };
+      return client;
+    } catch (e) {
+      console.error('Unable to initialise telemetry client', e);
+      return null;
+    }
+  }
+  return null;
+};
+
+let client = createClient();
+if (client) {
+  const onClose = () => {
+    client = createClient();
```

Comment:
Why client is created again?

Reply:
It is for reopening WebSocket connection if it dies out.